telemetry/csharp/AwsToolkit.Telemetry.Events/AwsToolkit.Telemetry.Events.csproj

```diff
@@ -16,7 +20,11 @@
 	</PropertyGroup>
 	<ItemGroup>
 	  <PackageReference Include="log4net" Version="2.0.8" />
+	  <PackageReference Include="NuGet.Build.Tasks.Pack" Version="5.2.0"/>
 	</ItemGroup>
+	<Target Name="PreBuild" BeforeTargets="PreBuildEvent">
+		<Exec Command="rem BUILD THE GENERATOR, THEN USE IT TO GENERATE CODE INTO THE EVENTS PROJECT&#xD;&#xA;dotnet build &quot;$(ProjectDir)..\AwsToolkit.Telemetry.Events.Generator\AwsToolkit.Telemetry.Events.Generator.csproj&quot;&#xD;&#xA;dotnet run --project &quot;$(ProjectDir)..\AwsToolkit.Telemetry.Events.Generator\AwsToolkit.Telemetry.Events.Generator.csproj&quot; -o $(ProjectDir) --namespace Amazon.AwsToolkit.Telemetry.Events.Generated&#xD;&#xA;" />
```

Comment:
`rem BUILD THE GENERATOR, THEN USE IT TO GENERATE CODE INTO THE EVENTS PROJECT`
What does this do? Should we put it as a comment above instead?

Reply:
rem is a command for adding remarks/comments to a batch/script file.

Reply:
The need to build and run the generator makes me think we should treat the source code/projects more as primitives that we can use to build up the package we want instead of using a prebuild event. Rather than baking this into the csproj file, what if a new (msbuild) proj file was created beside "telemetry/csharp/TelemetryClient.proj" which contains a task to produce the nuget package, which then performs these steps?  Everything else in this csproj file would likely remain as-is.